flask/flask-heartbeat-agent/build.gradle

```diff
@@ -0,0 +1,29 @@
+plugins {
+    id  'java-library'
+}
+
+import java.util.jar.Attributes
+
+evaluationDependsOn(":flask:flask-common")
```

Comment:
You shouldn't need this `evaluationDependsOn` for a simple module like this.

Reply:
I'll remove it

Reply:
I don't understand why those are needed [here](https://github.com/woggioni/corda-gradle-plugins/blob/flask/flask/build.gradle#L6-L7) though (build fails if I removed them)

Reply:
I think I got why.. it is because in that project, `project("flask-common")` is a `compileOnly` dependency and, as such, only the class directory is needed for the `compileClasspath` configuration (so the jar archive, that is subsequently used in the build, isn't built) while `implementation` adds the project jar artifact to the `runtimeClasspath` configuration, so it requires for it to be generated.

I still don't get why [this line](https://github.com/woggioni/corda-gradle-plugins/blob/flask/flask/build.gradle#L32) isn't enough to make Gradle understand it needs to invoke the `jar` task from the `flask-common` subproject first

Reply:
In `flask/build.gradle` you will need
```gradle
evaluationDependsOn("flask-common")
```
because you refer to it while eagerly configuring the `jar` task:
```gradle
jar {
    ...
    from(project("flask-common").tasks.named("jar").map { zipTree(it.outputs.files.singleFile) })
    ...
}
```
Gradle's `publish-plugin` seems to be forcing Gradle to configure that `jar` task too. However, you should be able to remove
```gradle
evaluationDependsOn("flask-launcher")
```
because the `copyLauncher` task is configured lazily.

Reply:
> I still don't get why [this line](https://github.com/woggioni/corda-gradle-plugins/blob/flask/flask/build.gradle#L32) isn't enough to make Gradle understand it needs to invoke the `jar` task from the `flask-common` subproject first

I think that line is the problem: you are asking for the `:flask-common` project's `jar` task before it knows that it _has_ a `jar` task. `:flask-common` is a child of `:flask` and must be evaluated _before_ `:flask` so that `:flask` can examine its contents.